### DISCUSSION TITLE
Retrying a task from inside task instance view

### DISCUSSION THREAD

**Discussion by ARtheboss** [apache/airflow#33523] [Ideas]
**Description**

Users should be able to clear a task along with all downstream tasks from the Task Instance page (especially the log subpage). In other words, we should be able to get a similar menu to the one that appears when you click on a task square from tree view on the Task Instance page.

**Use case / motivation**

This will make it easier to retry a task and will require less clicks (users won't have to go into the tree/graph page and then clear from there).

**Are you willing to submit a PR?**

Possibly

**Related Issues**

None

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Why not willing to implement it @ARtheboss ? Airflow is a community run project and a lot of users contribute to it, sounds like a simple change to implement? We are happy to guide you and help you with the contribution process.

**Comment by ARtheboss** [comment]
@potiuk Ok, I don't mind, just not sure how this works. Where are the relevant files for the web UI?

**Comment by potiuk** [comment]
No idea, I do not know that much of the UI. But I do not think it's as difficult - those are just HTML templates/javascript

**Comment by potiuk** [comment]
But there are others who might help

**Comment by uranusjr** [comment]
If you already know some Flask, what you need to do is to add a form in the UI that POST to a new view. That view should take the current task instance’s id, select it out, set its state to `UP_FOR_RETRY`, and then redirect back to the task instance view, similar to `TaskInstanceModelView.action_set_retry` (actually you can just post to that view, but the view is built against Flask-Appbuilder and manually POST-ing to it is a giant PITA that I strongly recommended against).

The form should look something like this:

```html
<form action="{{ url_for('name_of_view') }}" method="POST">
  <input type="hidden" name="csrf_token" value="{{ csrf_token() }}">
  <input type="hidden" name="dag_id" value="{{ dag_id }}">
  <input type="hidden" name="task_id" value="{{ task_id }}">
  <button type="submit">Retry Task</button>
</form>
```

**Comment by bbovenzi** [comment]
[Here](https://github.com/apache/airflow/blob/main/airflow/www/templates/airflow/task_instance.html) is the html file for the task instance page. I'd start there.

**Comment by zkan** [comment]
@potiuk May I take this issue, please?

**Comment by bbovenzi** [comment]
@zkan go for it!

**Comment by bbovenzi** [comment]
We added logs inside of the grid view and clear/mark actions are always available in that view.
We already plan to replace the entire legacy task instance view.
